cw_monero/lib/monero_subaddress_list.dart

```diff
@@ -86,4 +90,59 @@ abstract class MoneroSubaddressListBase with Store {
       rethrow;
     }
   }
+
+  Future<void> updateWithAutoGenerate({
+    required int accountIndex,
+    required String defaultLabel,
+    required List<String> usedAddresses,
+  }) async {
+    _usedAddresses.addAll(usedAddresses);
+    if (_isUpdating) {
+      return;
+    }
+
+    try {
+      _isUpdating = true;
+      refresh(accountIndex: accountIndex);
+      subaddresses.clear();
+      final newSubAddresses =
+          await _getAllUnusedAddresses(accountIndex: accountIndex, label: defaultLabel);
+      subaddresses.addAll(newSubAddresses);
+    } catch (e) {
+      rethrow;
+    } finally {
+      _isUpdating = false;
+    }
+  }
+
+  Future<List<Subaddress>> _getAllUnusedAddresses(
+      {required int accountIndex, required String label}) async {
+    final allAddresses = subaddress_list.getAllSubaddresses();
+
+    if (allAddresses.isEmpty || _usedAddresses.contains(allAddresses.last.getAddress())) {
+      final isAddressAdded = await _newSubaddress(accountIndex: accountIndex, label: label);
+      if (isAddressAdded) {
+        return await _getAllUnusedAddresses(accountIndex: accountIndex, label: label);
+      }
```

Comment:
1. naming is so misleading, it should be something like `isAddressUnused` or something
2. I believe the condition to enter the recursion loop is wrong, if the `_newSubaddress` returns `true` then no need to repeat the function, we can just join this recursive loop for `_newSubaddress` if the new subaddress is used